ui/app/adapters/job.js

```diff
@@ -34,11 +34,12 @@ export default class JobAdapter extends WatchableNamespaceIDs {
     return this.ajax(url, 'DELETE');
   }
 
-  parse(spec) {
+  parse(spec, jobVars) {
     const url = addToPath(this.urlForFindAll('job'), '/parse?namespace=*');
     return this.ajax(url, 'POST', {
       data: {
         JobHCL: spec,
+        JobVariables: jobVars,
```

Comment:
since we don't know what the eventual format expected for these will be, could you add a default value here? Whether that's an empty string, or a stringified object, or whatever the presumed format might be.

This'll help with rough edges if we try to call jobAdapter.parse from any other situation without jobVars included later, and as a side benefit we'll get some type hinting.

Reply:
This shouldn't have a default value other than `undefined` because it's an optional argument. Jobs do not need to include variables even if the specification was written in HCL2.

In order to think through that edge case, the only viable option I could think of was including defensive logic in the `model` method that is responsible for calling the `adapter` method here by ensuring variables are validly written in `JSON`. Otherwise, we'll dispatch the request and rely on the API to throw an error.